### ISSUE TITLE
Google Colab has started banning colabcode

### ISSUE THREAD

**Issue by ar8372** [abhishekkrthakur/colabcode#109]
colabcode is really an awesome tool, especially for someone who has very low local compute. But recently after the policy update of Google Colab (https://research.google.com/colaboratory/faq.html) , it has started banning use of tunneling. 

![colabcode](https://user-images.githubusercontent.com/91401599/170815642-b87b98ae-65ea-43c9-8498-4a5cb5f85d49.png)

Now Google Colab gets timeout after 10-15 minutes. Is there any solution to this @abhishekkrthakur ? any other alternative library, 

thanks.

**Comment by JapinderSandhu** [comment]
This is really unfortunate. Why would they do this? I think the same thing is happening with Kaggle now. Very sad. Ive been using this for so long I actually don't know what to do now. I cant import my code into cells this is so terrible. Importing files into Kaggle is a nightmare.

**Comment by kuro7766** [comment]
> This is really unfortunate. Why would they do this? I think the same thing is happening with Kaggle now. Very sad. Ive been using this for so long I actually don't know what to do now. I cant import my code into cells this is so terrible. Importing files into Kaggle is a nightmare.

Yes , kaggle editor is terrible. Is there any other ways to connect to kaggle jupyter server remotely?

**Comment by JapinderSandhu** [comment]
> 

I dont know, I tried looking into this method briefly.

https://towardsdatascience.com/how-to-kaggle-the-engineer-way-act-1-vs-code-containers-b3279970c029

https://code.visualstudio.com/docs/remote/containers

I dont know a lot about containers so im going to have to try and learn this method. 

I definitely want to avoid using kaggle cells. Kaggle cells are an absurd way to code.

**Comment by ar8372** [comment]
Hi @kuro7766 , @JapinderSandhu ,
I found a workaround for this, not the best but the closest I could get to vs code env on google-Colab.
First mount the drive where your code is available and then run below code. [Make sure you disable the add blocker if you are using brave]
```
!pip install colab-xterm
%load_ext colabxterm
%xterm
```
It will open a terminal for you. 
You can edit all your .py files or any other file by just double clicking on it and a tab will open for it. 
![terminal](https://user-images.githubusercontent.com/91401599/175159110-a2d557e4-7744-4283-a8ea-1449da1ed04c.png)

Another option is using GitHub Codespaces . GitHub provides with vs code env to write your code on fly.

**Comment by JapinderSandhu** [comment]
@ar8372 This is really terrible, the runtime on google colab disconnects randomly after 45 minutes. Google colab seems very unreliable compared to kaggle. 

this is not good

**Comment by JapinderSandhu** [comment]
> colabcode is really an awesome tool, especially for someone who has very low local compute. But recently after the policy update of Google Colab (https://research.google.com/colaboratory/faq.html) , it has started banning use of tunneling.
> 
> ![colabcode](https://user-images.githubusercontent.com/91401599/170815642-b87b98ae-65ea-43c9-8498-4a5cb5f85d49.png)
> 
> Now Google Colab gets timeout after 10-15 minutes. Is there any solution to this @abhishekkrthakur ? any other alternative library,
> 
> thanks.


UPDATE: S
After some digging and guidance this stackoverflow :
https://stackoverflow.com/questions/48350226/methods-for-using-git-with-google-colab

showed me a better method of using github and cloning my repository in the kaggle runtime using the !git clone command. I was unaware you could clone a github repository and run a python file in this method. I was even able to create an anaconda environment and run everything smoothly. So everything is running smoothly again :D <3 <3 :D

**Comment by qa13683954882** [comment]
google so Stupid. This does not make the paid feel safe,Instead, more choices for other GPU platforms,So why don't we choose another platform?
